commit 73185564fa795c484713f0205380068282ec0f3e
Author: Milian Wolff <mail@milianw.de>
Date:   Sat May 2 23:30:42 2020 +0200

    Actually remove temporary item repository directories
    
    The shutdown procedure was changed in commit c3c9ea8a3c, but that
    broke removal of temporary item repository directories. The repo
    was shutdown before the session is cleared, which will try to
    remove the dir. But it just marked the dir for removal, which should
    then happen in shutdown, but that was called already. So detect that
    and remove the dir directly then.

diff --git kdevplatform/serialization/itemrepositoryregistry.cpp kdevplatform/serialization/itemrepositoryregistry.cpp
index 0994d02725..234722ff20 100644
--- kdevplatform/serialization/itemrepositoryregistry.cpp
+++ kdevplatform/serialization/itemrepositoryregistry.cpp
@@ -111,6 +111,7 @@ class ItemRepositoryRegistryPrivate
 public:
     ItemRepositoryRegistry* m_owner;
     bool m_shallDelete;
+    bool m_wasShutdown;
     QString m_path;
     QMap<AbstractItemRepository*, AbstractRepositoryManager*> m_repositories;
     QMap<QString, QAtomicInt*> m_customCounters;
@@ -119,6 +120,7 @@ public:
     explicit ItemRepositoryRegistryPrivate(ItemRepositoryRegistry* owner)
         : m_owner(owner)
         , m_shallDelete(false)
+        , m_wasShutdown(false)
         , m_mutex(QMutex::Recursive)
     {
     }
@@ -158,6 +160,8 @@ void ItemRepositoryRegistry::initialize(const QString& repositoryPath)
         ///the actual repositories might get deleted later than the repository registry.
         m_self = new ItemRepositoryRegistry(repositoryPath);
     }
+    m_self->d_func()->m_wasShutdown = false;
+    m_self->d_func()->m_shallDelete = false;
 }
 
 ItemRepositoryRegistry* ItemRepositoryRegistry::self()
@@ -170,7 +174,7 @@ void ItemRepositoryRegistry::deleteRepositoryFromDisk(const QString& repositoryP
 {
     // Now, as we have only the global item-repository registry, assume that if and only if
     // the given session is ours, its cache path is used by the said global item-repository registry.
-    if (m_self && m_self->d_func()->m_path == repositoryPath) {
+    if (m_self && !m_self->d_func()->m_wasShutdown && m_self->d_func()->m_path == repositoryPath) {
         // remove later
         m_self->d_func()->m_shallDelete = true;
     } else {
@@ -426,5 +430,7 @@ void ItemRepositoryRegistry::shutdown()
     } else {
         QFile::remove(path + QLatin1String("/crash_counter"));
     }
+
+    d->m_wasShutdown = true;
 }
 }
